### ISSUE TITLE
"Promise was canceled" error when debugging in v1.6.2

### ISSUE THREAD

**Issue by drcallaway** [denoland/deno#8870]
It looks like a debugger regression was introduced in v1.6.2. I can debug from VS Code fine with v1.6.1 and earlier using a configuration like this:
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug Deno",
      "type": "pwa-node",
      "request": "launch",
      "cwd": "${workspaceFolder}",
      "runtimeExecutable": "deno",
      "program": "${file}",
      "runtimeArgs": [
        "run",
        "--inspect-brk",
        "--allow-all"
      ],
      "outputCapture": "std",
      "attachSimplePort": 9229
    }
  ]
}
```
However, upgrading to v1.6.2 results in this error when trying to connect the debugger:
```
Error processing launch: Error: Could not connect to debug target at http://127.0.0.1:9229: Promise was canceled
```
A temporary fix is to just roll back to the last working version:
```
deno upgrade --version 1.6.1
```

**Comment by lucacasonato** [comment]
Possibly related to #8813?

cc @bartlomieju  @lperlaki

**Comment by kitsonk** [comment]
The PR that was merged has been reverted, as it logged `hello` repeatedly to `stdout` and did not tear itself down, as well as introduced openssl by depending on reqwest directly instead of using it from `deno_fetch`.  The PR needs to be reworked before it can be landed again.

**Comment by bartlomieju** [comment]
Fix relanded in #8986
